server/bin/pbench-index.py

```diff
@@ -40,6 +48,236 @@
+            _pid = pid if pid else os.getpid()
+            _group_id = group_id if group_id else os.getgid()
+            _user_id = user_id if user_id else os.getuid()
+        self.generated_by = dict(
+            [
+                ("commit_id", self.config.COMMIT_ID),
+                ("group_id", _group_id),
+                ("hostname", _hostname),
+                ("pid", _pid),
+                ("user_id", _user_id),
+                ("version", version if version else ""),
+            ]
+        )
+        # The "tracking_id" is the final MD5 hash of the first document
+        # indexed via the `post_status()` method.
+        self.tracking_id = None
+        try:
+            self.idx_prefix = config.get("Indexing", "index_prefix")
+        except (NoOptionError, NoSectionError):
+            # No index prefix so reporting will be performed via logging.
+            self.idx_prefix = None
+            self.es = None
+        else:
+            if es is None:
+                try:
+                    self.es = get_es(config, self.logger)
+                except Exception:
+                    self.logger.exception(
+                        "Unexpected failure fetching" " Elasticsearch configuration"
+                    )
+                    # If we don't have an Elasticsearch configuration just use
+                    # None to indicate logging should be used instead.
+                    self.es = None
+            else:
+                self.es = es
+        if templates is not None:
+            self.templates = templates
+        else:
+            self.templates = PbenchTemplates(
+                self.config.BINDIR, self.idx_prefix, self.logger
+            )
+
+    def init_report_template(self):
+        """Setup the Elasticsearch templates needed for properly indexing
+        report documents. This is only needed by non-'pbench-index' use cases.
+        """
+        if self.es is None:
+            return
+        self.templates.update_templates(self.es, "server-reports")
+
+    @staticmethod
+    def _make_json_payload(source):
+        """Given a source dictionary, return its ID, and a formatted JSON
+        payload.
+        """
+        the_bytes = json.dumps(source, sort_keys=True).encode("utf-8")
+        source_id = hashlib.md5(the_bytes).hexdigest()
+        return source, source_id, the_bytes.decode("utf-8")
+
+    def _gen_json_payload(self, base_source, file_to_index):
+        """Generate a series of JSON documents to be indexed, where the text
+        payload has a maximum size of 50 MB.
+        """
+        total_size = os.path.getsize(file_to_index)
+        number_of_chunks = int(math.ceil(total_size / self._CHUNK_SIZE))
+        chunk_id = 0
+        with open(file_to_index, "r") as fp:
+            EOF = False
+            while not EOF:
+                chunk_id += 1
+                text = fp.read(self._CHUNK_SIZE)
+                if not text:
+                    EOF = True
```

Comment:
Since you're going to the trouble of doing an explicit check, why not just `break` here and get rid of the `EOF` flag?  (And de-indent the else block.)

Reply:
Lift-n-shift code.